### DISCUSSION TITLE
preview files list

### DISCUSSION THREAD

**Discussion by ramcq** [borgbase/vorta#839] [Ideas]
**Is your feature request related to a problem? Please describe.**
When setting up new backup profiles, one thing I look out for when setting up the exclude patterns is whether there is something included that should not be - caches, large downloaded files, special filesystem mounts, etc. Sometimes if I make a mistake, I only notice part way through the backup when I see the status/info that a file I don't want is being backed up, and at this point the only option I have is to cancel and start the backup over again, losing a lot of time.

**Describe the solution you'd like**
It would be great if Vorta offered a way to do a dry run and list the files that were included in the backup, so that I could inspect for files which I didn't want to include in the backup (or if the exclude pattern was done wrong). If it could be sorted by size and mtime, that would also allow easy inspection for large files, and recently-modified ones (eg cache/scratch data which could cause noise/churn in the backups).

**Describe alternatives you've considered**
Because I don't know how to set up a borg commandline that matches what is configured via the UI, I'm not sure if there is an easy alternative to Vorta doing this for me. Perhaps there is but if you could let me know that would be superb! Thanks for BorgBackup, Vorta and BorgBase - I'm really happy so far.

**Comment by archont00** [comment]
On Archive tab, there is already a "Diff" command, which would show you a filesystem tree with changes (additions, removals, modifications) between any two archives (snapshots) and file sizes. It can't be exported though and cannot be sorted by size.

**Comment by archont00** [comment]
AFAIK `borg create` cannot show file sizes. If you are on linux, you can try this to get the file list from `borg create` sorted by size (in kB):
```
(while IFS=$'\n' read -r line; do ls --directory --size --kibibytes -x $line; done | sort --reverse --numeric-sort) < <(borg create --dry-run --list --exclude /some/pattern/ /my/repo::test /var/cache/pacman 2>&1 | cut -d' ' -f2)
```
which produces something like this:
```
130736 /var/cache/pacman/pkg/go-2:1.15.3-1-x86_64.pkg.tar.zst
130456 /var/cache/pacman/pkg/go-2:1.14.7-1-x86_64.pkg.tar.zst
130448 /var/cache/pacman/pkg/go-2:1.14.4-1-x86_64.pkg.tar.zst
130244 /var/cache/pacman/pkg/go-2:1.14.1-1-x86_64.pkg.tar.zst
127464 /var/cache/pacman/pkg/linux-firmware-20200916.00a84c5-1-any.pkg.tar.zst
125384 /var/cache/pacman/pkg/go-2:1.13.5-1-x86_64.pkg.tar.xz
125356 /var/cache/pacman/pkg/go-2:1.13.4-1-x86_64.pkg.tar.xz
```

**Comment by m3nu** [comment]
We just merged this PR: https://github.com/borgbase/vorta/pull/617 by @XXXBold.

This may partially solve your issue. You don't see individual files though. And Vorta probably should re-implement `ncdu` or similar tools.

We do plan on applying exclusions to those size estimates at some point. Then you could see if the right files are excluded.
